File: lecture_007(Strings)/l_002_printcharandsubstrings.cpp
#include<iostream>
#include<string>
using namespace std;

void displaychar(string s) {
    for(int i=0; i<s.size(); i++) {
        cout<<s[i]<<" ";
    }
    cout<<endl;
}
void displaysubstrings(string s) {
    for(int i=0; i<s.size(); i++) {
        for(int j=1; j<=s.size()-i; j++) {
            cout<<s.substr(i,j)<<endl;
        }
    }
}
int main() {
    string s;
    cin>>s;
    cout<<s<<endl;
    displaychar(s);
    //displaysubstrings(s);

    return 0;
}

File: lecture_007(Strings)/l_002_printcharandsubstrings.java
import java.util.Scanner;
public class l_002_printcharandsubstrings {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        String s = scn.nextLine();
        //System.out.println(s);
        displayChar(s);
        displaySubstrings(s);
    }
    public static void displayChar(String s) {
        for(int i=0; i<s.length(); i++) {
            System.out.print(s.charAt(i)+" ");
        }
        System.out.println();
    }
    public static void displaySubstrings(String s) {
        for(int i=0; i<s.length(); i++) {
            for(int j=i+1; j<=s.length(); j++) {
                System.out.println(s.substring(i,j));
            }
        }
    }
}